mm2src/coins/utxo/utxo_balance_events.rs

```diff
@@ -172,3 +149,30 @@ impl EventStreamer for UtxoBalanceEventStreamer {
         }
     }
 }
+
+async fn subscribe_to_addresses(utxo: &UtxoCoinFields, addresses: HashSet<Address>) -> HashMap<String, Address> {
+    match utxo.rpc_client.clone() {
+        UtxoRpcClientEnum::Electrum(client) => {
+            // Collect the scrpithash for every address into a map.
+            let scripthash_to_address_map = addresses
+                .into_iter()
+                .filter_map(|address| {
+                    let scripthash = address_to_scripthash(&address)
+                        .map_err(|e| log::error!("Failed to get scripthash for address {address}: {e}"))
+                        .ok()?;
+                    Some((scripthash, address))
+                })
+                .collect();
+            // Add these subscriptions to the connection manager. It will choose whatever connections
+            // it sees fit to subscribe each of these addresses to.
+            client
+                .connection_manager
+                .add_subscriptions(&scripthash_to_address_map)
+                .await;
+            scripthash_to_address_map
+        },
+        UtxoRpcClientEnum::Native(_) => {
+            unreachable!("The caller of this func checked that the RPC client is electrum. Native client isn't supported for balance streaming.")
+        },
```

Comment:
Let's just return an empty hashmap in such case, I don't like panics in the codebase specially in something that is rarely used and might not get caught easily in testing or production.

Reply:
agreed. done here: [7ff638b](https://github.com/KomodoPlatform/komodo-defi-framework/pull/2431/commits/7ff638b8059b3ac7042d38578826914740407b84)